_benchmark/creating-a-custom-workload.md

```diff
@@ -0,0 +1,374 @@
+--workload="<WORKLOAD NAME>" \
+--target-hosts="<CLUSTER ENDPOINT>" \
+--client-options="basic_auth_user:'<USERNAME>',basic_auth_password:'<PASSWORD>'" \
+--indices="<INDICES TO GENERATE WORKLOAD FROM>" \
+--output-path="<LOCAL DIRECTORY PATH TO STORE WORKLOAD>"
+```
+
+Replace the following options in the preceding example with information specific to your existing cluster:
+
+- `--workload`: A custom name for your custom workload.
+- `--target-hosts:` A comma-separated list of host:port pairs that workload targets.
+- `--client-options`: The basic auth client options OpenSearch Benchmark uses to access the cluster.
+- `--indices`: One or more indexes inside your OpenSearch cluster which contain data.
+- `--out-path`: The directory from which the workload outputs Benchmark metrics.
+
+The following example response creates a workload named `movies` from a cluster with an index named `movies-info`. The `movies-info` index contains over 2000 documents.
+
+```
+   ____                  _____                      __       ____                  __                         __
+  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
+ / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
+/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
+\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
+    /_/
+
+[INFO] You did not provide an explicit timeout in the client options. Assuming default of 10 seconds.
+[INFO] Connected to OpenSearch cluster [380d8fd64dd85b5f77c0ad81b0799e1e] version [1.1.0].
+
+Extracting documents for index [movies] for test mode...      1000/1000 docs [100.0% done]
+Extracting documents for index [movies]...                    2000/2000 docs [100.0% done]
+
+[INFO] Workload movies has been created. Run it with: opensearch-benchmark --workload-path=/Users/hoangia/Desktop/workloads/movies
+
+-------------------------------
+[INFO] SUCCESS (took 2 seconds)
+-------------------------------
+```
+
+As part of workload creation, OpenSearch Benchmark generates the following files. You can access them in directory specified by the `--out-path` option.
+
+- `workload.json`: Contains general workload specifications.
+- `<index>.json`: Contains mappings and settings for the extracted indexes.
+- `<index>-documents.json`: Contains the sources of every document from the extracted indexes. Any sources suffixed with `-1k` encompasses only a fraction of the document corpus of the workload and will primarily run in the text mode.
+
+By default, OpenSearch Benchmark does not contain a reference to generate queries. Because you know your own data the best, we recommend adding a `match_all` query to your `workload.json` file based on your index's specifications, as shown in the following example query:
```

Comment:
Nit: I suggested the match_all query since it's the most simple query. It might be better to restructure this sentence:
"Since you have the best understanding of their data, we recommend that you first structure queries based on the index's specifications and then add them to `workload.json`. When inserting the queries into `workload.json`, feel free to model it after the following `match_all` query example"

Reply:
Looks good! Nit: "example of a query"